website/content/en/docs/building-operators/golang/webhook.md

```diff
@@ -0,0 +1,185 @@
+
+```sh
+├── Dockerfile
+├── Makefile
+├── PROJECT
+├── api
+│   └── v1alpha1
+│       ├── memcached_webhook.go
+│       ├── webhook_suite_test.go
+├── config
+│   ├── certmanager
+│   │   ├── certificate.yaml
+│   │   ├── kustomization.yaml
+│   │   └── kustomizeconfig.yaml
+│   ├── default
+│   │   ├── manager_webhook_patch.yaml
+│   │   └── webhookcainjection_patch.yaml
+│   └── webhook
+│       ├── kustomization.yaml
+│       ├── kustomizeconfig.yaml
+│       └── service.yaml
+├── go.mod
+├── go.sum
+└── main.go
+```
+
+The scaffolded file `api/v1alpha1/memcached_webhook.go` has method signatures which need to be implemented for the validation webhook.
+
+Following this, there are a few steps which need to be done in your operator project to enable webhhoks. This will involve:
+
+1. Implementing the required methods for Validating or Mutating webhook in <kind>_webhook.go. An example of such implementation is provided [here](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html).
+
+2. Uncommenting sections in config/default/kustomization.yaml to enable webhook and cert-manager configuration through kustomize. Cert-manager (or any third party solution) can be used to provision certificates for webhook server. This is explained in detail here.
+
+### Generate webhook manifests and enable webhook deployment 
+
+Once your webhooks are implemented, all that’s left is to create the WebhookConfiguration manifests required to register your webhooks with Kubernetes:
+
+```sh
+$ make manifests
+```
+
+You will need to enable cert-manager and webhook deployment in order to deploy these webhooks properly. 
+To do so, edit the `config/default/kustomize.yaml` and uncomment the sections marked by `[WEBHOOK]` and `[CERTMANAGER]` comments. 
+More detail on this step can be found in the [Kubebuilder documentation](https://book.kubebuilder.io/cronjob-tutorial/running-webhook.html).
+
+### Update main.go so that running locally works 
+
+Refer upstream [Kubebuilder doc](https://book.kubebuilder.io/cronjob-tutorial/main-revisited.html?highlight=enable_webhooks#you-said-something-about-main) for more details.
+
+## Run your operator and webhooks 
+
+There are two ways to test your operator project with webhooks.
+
+#### Run locally
+
+Technically, the webhooks can be run locally, but for it to work you need to generate certificates for the webhook server and store them at /tmp/k8s-webhook-server/serving-certs/tls.{crt,key}. For more details about running webhook locally, refer [here](https://book.kubebuilder.io/cronjob-tutorial/running.html#running-webhooks-locally).
+
+#### Run as a Deployment inside the cluster
+
+Adding webhooks does not alter deploying your operator. For instructions on deploying your operator into a cluster, refer to the [tutorial](https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/#2-run-as-a-deployment-inside-the-cluster).
+
+## Exercise your webhook
```

Comment:
This entire section is not needed either. We have not explained that the validation implementation here is on number of memcached pods. We have directed users upstream for implementation example. Hence having this example would be out of the blue for users. This can be removed.

Reply:
Yes, agree. Removed it completely.